### ISSUE TITLE
Implicit Maximum Likelihood vs "Classic" VAE loss

### ISSUE THREAD

**Issue by vincentmaye** [jsll/pytorch_6dof-graspnet#10]
Hey Jens,

thank you for the really great work! I was wondering if you could share some experiences on training the VAE in the "GAN formulation" using Implicit MLE estimation vs. using the original VAE loss. Did you have issues with mode collapse and did IMLE solve these for you? Did you experience any drawbacks from IMLE?

**Comment by jsll** [comment]
Hi Vincent,

thanks for those kind words. As far as I remember, IMLE reached a lower total loss in fewer epochs which were the main reasons I chose to use it instead of regular VAE loss. I have not experimented anything with mode collapse, but as IMLE was developed do combat such behavior I would hypothesize that it at least will not make it more prone to mode collapse.

**Comment by vincentmaye** [comment]
Great, that's really valuable, thanks for the info! So in terms of training speed, you recommend the IMLE. Do you have some intuition regarding the accuracy of poses sampled from the generator: Did IMLE also outperform here?

**Comment by jsll** [comment]
We did not use IMLE to train the generator as it is tasked to auto-encode the input.  So we do not have a GAN setting here. If you want to use the IMLE to train the generator you need to change how it is trained
